src/Database/Type/DateTimeType.php

```diff
@@ -206,9 +207,17 @@ public function marshal($value)
             $value['minute'],
             $value['second']
         );
-        $tz = isset($value['timezone']) ? $value['timezone'] : null;
+        $tz = isset($value['timezone']) ? $value['timezone'] : Configure::read('App.defaultOutputTimezone');
 
-        return new $class($format, $tz);
+        /* @var \Cake\I18n\Time $time */
+        $time = new $class($format, $tz);
+
+        $internalTimezone = date_default_timezone_get();
+        if ($tz && $tz !== $internalTimezone) {
+            $time = $time->timezone($internalTimezone);
```

Comment:
Doesn't this change mean the timezone should be again set to default timezone before converting the object to string in `toDatabase()` method?

Reply:
My tests seemed to yield the right conversion, so maybe we do not? But I am happy to be proven wrong.